src/app/pages/vehicle.cpp

```diff
@@ -141,17 +144,32 @@ QWidget *VehiclePage::can_bus_toggle_row()
     QWidget *widget = new QWidget(this);
     QHBoxLayout *layout = new QHBoxLayout(widget);
 
-    QLabel *label = new QLabel("CAN Bus", widget);
+    QLabel *label = new QLabel("Interface", widget);
     layout->addWidget(label, 1);
 
-    Switch *toggle = new Switch(widget);
-    toggle->scale(this->config->get_scale());
-    toggle->setChecked(this->config->get_vehicle_can_bus());
-    connect(this->config, &Config::scale_changed, [toggle](double scale) { toggle->scale(scale); });
-    connect(toggle, &Switch::stateChanged, [this](bool state) {
-        this->config->set_vehicle_can_bus(state);
-    });
-    layout->addWidget(toggle, 1, Qt::AlignHCenter);
+    QGroupBox *group = new QGroupBox();
+    QVBoxLayout *group_layout = new QVBoxLayout(group);
+
+
+    QRadioButton *socketcan_button = new QRadioButton("SocketCAN", group);
+    socketcan_button->setChecked(this->config->get_vehicle_can_bus());
+    connect(socketcan_button, &QRadioButton::clicked,
+            [config = this->config]() { 
+                config->set_vehicle_can_bus(true); 
+                emit config->vehicle_can_bus_changed(true);
```

Comment:
you shouldn't need to emit this signal here, it's already done in the call to `Config::set_vehicle_can_bus` (same for below)

Reply:
Thanks for the reminder - I thought I might have just overlooked it the first time around